src/services/loginService.test.ts

```diff
@@ -0,0 +1,42 @@
+import { AzureLoginService } from "./loginService";
+
+describe('Login Service', () => {
+  beforeAll(() => {
+    // Because the functions we use for authentication are exported
+    // as functions from their respective modules 
+    // (open, interactiveLoginWithAuthResponse and 
+    // loginWithServicePrincipalSecretWithAuthResponse),
+    // it is extremely difficult to mock their functionality.
+    // As a workaround, they have been placed in the thinnest possible
+    // functions within the Azure Login service, which we will
+    // use to make assertions on the functionality of the login service itself
+    AzureLoginService.interactiveLogin = jest.fn();
```

Comment:
Another idea would be to mock the whole incoming module using `jest.mock`.  Virtually the same behavior so I am fine with this approach.